libs/cardano-data/src/Data/ListMap.hs

```diff
@@ -143,3 +150,21 @@ filter f (ListMap xs) = ListMap $ Pre.filter (uncurry f) xs
 
 toMap :: Ord k => ListMap k v -> Map.Map k v
 toMap (ListMap xs) = Map.fromList xs
+
+fromMap :: Map.Map k v -> ListMap k v
+fromMap = ListMap . Map.toList
+
+mapKeys :: (k1 -> k2) -> ListMap k1 a -> ListMap k2 a
+mapKeys f = ListMap . fmap (first f) . unListMap
+
+map :: Ord k => (a -> v) -> ListMap k a -> ListMap k v
+map f = ListMap . Map.toList . Map.map f . Map.fromList . unListMap
```

Comment:
@Soupstraw I saw that thing as well, do you wanna submit a PR with this fix, considering this one has already been merged. Using list comprehensions would very a clean solution.

Reply:
In fact, as defined here, it strictly speaking is not even a correct `map`, since it also removes duplicates and affects the order of elements. Which is not an invariant we maintain in this implementation, as far as I am aware